lib/config/param.c

```diff
@@ -443,6 +445,40 @@ param_default_converter(const struct param_spec *ps, const cJSON *node, void *va
             *(uint64_t *)value = tmp;
             break;
         }
+        case PARAM_TYPE_SIZE: {
+            char *end;
+            char buf[32];
+            size_t tmp;
+            double to_conv;
+
+            assert(!(ps->ps_flags & PARAM_FLAG_NULLABLE));
+            if (!cJSON_IsNumber(node)) {
+                log_err("Value of %s must be a number", ps->ps_name);
+                return false;
+            }
+
+            to_conv = cJSON_GetNumberValue(node);
+            if (!IS_WHOLE(to_conv)) {
+                log_err("Value of %s must be a whole number", ps->ps_name);
+                return false;
+            }
+
+            /* Converting a double to a 64-bit size_t is extremely difficult.
+             * Confirm that the double is a whole number, convert it to a
+             * string, and parse the string to a size_t.
+             */
+            snprintf(buf, sizeof(buf), "%.0f", to_conv);
+            tmp = strtoull(buf, &end, 10);
+            if (to_conv < 0 || *end != '\0') {
+                log_err(
+                    "Value of %s must be greater than or equal to 0 and less than or equal to %lu",
+                    ps->ps_name,
+                    SIZE_MAX);
+                return false;
+            }
+            *(size_t *)value = tmp;
+            break;
+        }
```

Comment:
I still don't know why we need to do this..

FWIW, in order to use the strtoul() type functions correctly you need to set errno=0 before calling, then check errno after the call, then check to see if return code is ULONG_MAX and if see then check errno...
I don't think I'd trust that end is valid in that case...

Reply:
It can all go away when cJSON is removed.

Hopefully fixed the error checking in next push.

Reply:
Cool, btw, shouldn't you use %lf in your snprintf?

Reply:
Whoops, fixed.